### DISCUSSION TITLE
CEBRA and feature interpretation

### DISCUSSION THREAD

**Discussion by YuhaoHuangMD** [AdaptiveMotorControlLab/CEBRA#19] [Q&A] [answered]
Hi CEBRA team!

Fantastic work on the paper/method. I was wondering if CEBRA provides any method of relating the underlying predictive latent space back to the feature space? For example, given multi-site brain recordings (N channels), and multiple ephys features (power, connectivity x freq), and associated behavior - using CEBRA could produce a consistent latent behavior embedding, but is there a intuitive way to relating the latent embeddings back to the feature space such that we understand what features are related to behavior?

**Comment by stes** [comment, marked as answer]
Hi @YuhaoHuangMD , thanks for your question!

One simple first option could be to train a CEBRA behavior embedding, and then try to decode the features of interest (we provide a few kNN decoders in the package that could be a good place to start).

There might be other options, like fitting embeddings both with the behavior and the features (note, the "behavior" labels can actually be features you computed yourself), and comparing consistency across them, etc

But from what you wrote it seems like starting with decoding from a cebra behavior embedding could be a good start.

Does that address your question?

Maybe to give a more refined answer, what shapes would your data matrices (recordings, features, behavior) have in this example?

**Reply by YuhaoHuangMD** [reply]
Thanks for your reply! I see, you would recommend using the CEBRA behavior latent embedding to see which input features could be decoded with the less error, and therefore would represent the most weight to the latent embedding? 

I know that in other generative / predictive models like VAE, you could potentially train to minimize both predictive loss (behavior based) and reconstructive loss (feature based), and in the final output, you can see which latent embeddings contributed to the final feature construction, thereby allowing interrogation of the input feature importance to the model. From my understanding, CEBRA does not have a generative aspect / or use reconstructive loss, is that correct?

**Reply by MMathisLab** [reply]
Correct, we specifically designed it not to have a generative model (see introduction of paper and detailed Suppl. text for reasons)
